include/deal.II/non_matching/mapping_info.h

```diff
@@ -656,9 +656,19 @@ namespace NonMatching
     /**
      * A pointer to the internal data of the underlying mapping.
      */
-    std::shared_ptr<typename Mapping<dim, spacedim>::InternalDataBase>
+    std::unique_ptr<typename Mapping<dim, spacedim>::InternalDataBase>
       internal_mapping_data;
 
+    /**
+     * Quadrature object that is used for the reinit(ArrayView) function.
+     */
+    Quadrature<dim> quadrature;
+
+    /**
+     * An array containing the data fields of this class.
```

Comment:
Can you clarify here that this is only a temporary helper to ask Mapping for the requested information for one cell?